lib/ui/window/pointer_data_packet_converter.cc

```diff
@@ -0,0 +1,287 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/window/pointer_data_packet_converter.h"
+#include "flutter/fml/logging.h"
+
+#include <string.h>
+
+namespace flutter {
+
+PointerDataPacketConverter::PointerDataPacketConverter() : pointer_(0) {}
+
+PointerDataPacketConverter::~PointerDataPacketConverter() = default;
+
+std::unique_ptr<PointerDataPacket> PointerDataPacketConverter::Convert(
+    std::unique_ptr<PointerDataPacket> packet) {
+  size_t kBytesPerPointerData = kPointerDataFieldCount * kBytesPerField;
+  auto buffer = packet->data();
+  size_t buffer_length = buffer.size();
+
+  std::vector<PointerData> converted_pointers;
+  // Converts each pointer data in the buffer and stores it in the
+  // converted_pointers.
+  for (size_t i = 0; i < buffer_length / kBytesPerPointerData; i++) {
+    PointerData pointer_data =
+        *(reinterpret_cast<PointerData*>(&buffer[i * kBytesPerPointerData]));
```

Comment:
@chinmaygarde Is this the right way to go? I am not sure what is the standard way to use copy constructor here

Reply:
I apologize for my previous comment. I didn't see that the data structure was packed and you were reading from an offset that may not be properly aligned. Your first attempt was better. Please use the memcpy. Sorry for the confusion.